### DISCUSSION TITLE
gunicorn, uvicorn, and --factory

### DISCUSSION THREAD

**Discussion by daBrado** [Kludex/uvicorn#1810] [Q&A] [answered]
When I use a factory-style app with Uvicorn + Gunicorn, part of the startup gives the warning:

    [WARNING] ASGI app factory detected. Using it, but please consider setting the --factory flag explicitly.

However, this presumably is referring to the `--factory` flag to uvicorn, as there isn't such a flag for gunicorn.

Is there a way to enable the factory option for Uvicorn when running as a Gunicorn worker?

I tried using an environment variable but I guess that only gets parsed if using the `uvicorn` CLI.  The only way I knew to get it to not emit a warning was to subclass the `UvicornH11Worker` with a `CONFIG_KWARGS` modified with `"factory":"true"`, but it would be nice to be able to just use the command line + environment since otherwise I have no need to create a subclass.

**Comment by Kludex** [comment, marked as answer]
Use:
```bash
gunicorn "main:make_app()" --worker-class uvicorn.workers.UvicornWorker
```

Instead of:
```bash
gunicorn main:make_app --worker-class uvicorn.workers.UvicornWorker
```

Ref.: https://stackoverflow.com/questions/25319690/how-do-i-run-a-flask-app-in-gunicorn-if-i-used-the-application-factory-pattern

**Comment by daBrado** [comment]
Nice, thanks!
